### PR TITLE
Add Login page + Password Change Page + add auth providers

### PR DESCRIPTION
**Pull request by ebenezer-melkamu**
### Modifications 

1. Added Login page
2. Added Change password page
3. Added new `Auth Provider methods` 
4. Modified the existing `Data Provider Methods`

### CODE REVIEW

File: src/common/dataProvider.ts [hunk 1]
```diff
@@ -16,60 +14,62 @@ import { DataProvider } from 'ra-core';
...
+    field = field === "id" ? pkDictionary[resource] : field;
     const ordering = order === "ASC" ? `${field}` : `-${field}`;
 
     //since the soul api requires comma separated filters, remove the quotes and curly braces from the filter
     const filter = JSON.stringify(params.filter).replace(/[{} ""]/g, "");
 
-    //Add the _extend query to get many data by using a foreign key 
-    let extend = undefined 
+    //Add the _extend query to get many data by using a foreign key
+    let extend = undefined;
```
Comment (IanMayo) [inline src/common/dataProvider.ts]: @AbegaM - can we remove the `chinook` specific code (`artistId`) on the following lines from the `dataProvider`?

Comment (ebenezer-melkamu) [reply src/common/dataProvider.ts]: Sure @IanMayo I will remove it

**Comment by IanMayo** [comment]
I'm not fully sure if this is an issue or not, and I'm meant to be reviewing the `xxxProviders` rather than your test app, but...  When I start the app, I'm already logged in as the new user - even with cookies cleared.

I have to logout to get the login dialog.

Hmm, I was also expecting Soul to add the auth related tables to `chinook.db` when first run, but it isn't.

I used `logout` from the top-right context menu, tried to login (deliberately using wrong password) and got a Soul error:
image

Aah, hold on.  As the image shows, we're not running in auth mode.  Aaah, I'm running `dev:soul-client`, which doesn't have auth enabled.  So. _maybe_ Soul shouldn't fall over when `login` end-point is called if `auth` is off.  Maybe return error indicating `login end-point not active if auth disabled`.

**Comment by IanMayo** [comment]
Another issue. When I'm logged in, and try to `Change Password` the server throws an error:
image

This time I actually pasted the current password into all 3 fields, the previous time I used a different new password.

**Review by IanMayo — APPROVED** [review]
Tested and working

**Pull request merged**